bridge/docs/bridging.md

```diff
@@ -34,3 +34,91 @@ Again, a withdrawfee of 1 TFT will be taken, so make sure you send a larger amou
 The amount withdrawn from TF Chain will be sent to your Stellar wallet.
 
 Example: ![swap_to_stellar](swap_to_stellar.png)
+
+## Deeper look at how the TFChain Bridge works
+
+### STELLAR -> TFCHAIN (Lock-and-Mint flow)
+In this section, we look into the details of transferring TFT from a Stellar Account to a TFChain Account. 
+
+1. A transaction is received on the bridge Stellar account and witnessed by a bridge validator..
+
+2. The transaction information undergoes validation. If it fails, a refund is issued (sent back to the stellar source account). now, We assume that the validation has passed, but we will examine the refund flow in the next section.
+
+3. A mint is proposed by calling `propose_or_vote_mint_transaction` extrinsic on the TFTBridgeModule in tfchain. This extrinsic inserts a new `MintTransaction` in `MintTransactions` storage that includes the `amount`, `target`, `block`, `votes`, and emits a `MintTransactionProposed` event. The mint is considered processed by the bridge side at this point.
+
+4. Other bridge validators also votes for that proposal by calling same extrinsic. If the `MintTransaction` exists in `MintTransactions` storage already, the extrinsic increments the `votes` count for the specified `MintTransaction` and emit `MintTransactionVoted` event.
+
+5. From the TFChain side, if the majority (more than the half) of bridge validators agree on the transaction, tokens are minted to the target address. This check happens every time the `propose_or_vote_mint_transaction` extrinsic is executed by validator call. Then, the transaction is removed from bridge pallet `MintTransactions` storage and added to `ExecutedMintTransactions`. Finally, a `MintCompleted` event is emitted.
+
+#### Overview of the TFChain Minting events
+1. `tftBridgeModule.MintTransactionProposed`: A bridge validator has proposed a mint transaction upon witness a stellar deposit.
+2. `tftBridgeModule.MintTransactionVoted`: Other bridge validators witness same stellar deposit and voted for the mint proposal.
+3. `tftBridgeModule.MintCompleted`: Enough bridge validators' votes was collected and the tokens minted to the target address successfully.
+
+#### When a Refund-on-Stellar occurs?
+
+A refund on Stellar occurs when one of the following conditions is met:
+
+- The deposited amount is lower than the deposit fee.
+- The memo message is empty.
+- The transaction contains more than one payment.
+- The memo is not formatted correctly.
+- The grid type is not supported (not one of grid, farm, node, or entity) or not found.
+
+### STELLAR -> TFCHAIN (Refund-on-Stellar flow)
+In this section, we look into the details of what happens when the a Stellar deposit can not be processed sue to a validation problem.
+
+1. A transaction is received on the bridge Stellar account and witnessed by a bridge validator.
```

Comment:
Same as previous section I would rephrase a bit for better unsderstanding

1. A transaction is received on the bridge Stellar account (a.k.a vault) and monitored by bridge validators (deamons).

2. Each time such transaction event is reported by a bridge validator it undergoes some validation. Starting from this point, we assume that the validation has failed because because of one of the violations mentioned in the previous section so a refund flow is initiated.

3. The first bridge validator reporting a violation will initiate the refund by calling TFTBridgeModule `create_refund_transaction_or_add_sig` extrinsic to propose a `RefundTransaction`, to store the details in `RefundTransactions` storage map alongside with its signature and to emit `RefundTransactionsignatureAdded` and `RefundTransactionCreated` events.

4. Other bridge validators that report the transaction later also provides their signature for that refund transaction proposal by calling same extrinsic.

5. If ...

6. 3. The first bridge validator reporting the `RefundTransactionReady` event will handle it and query TFChain storage for ...